src/rules/promiseFunctionAsyncRule.ts

```diff
@@ -29,10 +51,30 @@ export class Rule extends Lint.Rules.TypedRule {
             throwing an Error object. In contrast, non-\`async\` \`Promise\`-returning functions
             are technically capable of either. This practice removes a requirement for consuming
             code to handle both cases.
+
+            Several arguments may be optionally provided:
+
+            * \`"${OPTION_FUNCTION_DECLARATION}"\` check function declarations.
+            * \`"${OPTION_FUNCTION_EXPRESSION}"\` check function expressions.
+            * \`"${OPTION_ARROW_FUNCTION}"\` check arrow functions.
+            * \`"${OPTION_METHOD_DECLARATION}"\` check method declarations.
         `,
         optionsDescription: "Not configurable.",
-        options: null,
-        optionExamples: [true],
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+                enum: [
+                    OPTION_FUNCTION_DECLARATION,
+                    OPTION_FUNCTION_EXPRESSION,
+                    OPTION_ARROW_FUNCTION,
+                    OPTION_METHOD_DECLARATION,
+                ],
+            },
+            minLength: 0,
+            maxLength: 4,
+        },
+        optionExamples: [[true, OPTION_FUNCTION_DECLARATION, OPTION_METHOD_DECLARATION]],
```

Comment:
maybe we can add two examples here, the one that you added and the other `true`?